src/managers/GuildManager.js

```diff
@@ -190,8 +191,15 @@ class GuildManager extends CachedManager {
       if (channel.type) channel.type = ChannelTypes[channel.type.toUpperCase()];
```

Comment:
```suggestion
      if (channel.type) channel.type = ChannelTypes[channel.type.toUpperCase()];
      if (channel.type) channel.type = typeof channel.type === 'number' ? channel.type : ChannelTypes[channel.type];
```

Will have to update this too to accept a number:

https://github.com/discordjs/discord.js/blob/839974ca432c70d57e82e9b47334f743e0c20281/src/managers/GuildManager.js#L87